src/cromshell/execution_status_count/command.py

```diff
@@ -0,0 +1,251 @@
+        # If task has a key called 'subworkflowMetadata' in its first (zero) element
+        # (shard) and expand_sub_workflow parameter is set to true then rerun this
+        # function on that subworkflow
+        if (
+                "subWorkflowMetadata" in workflow_metadata["calls"][task][0]
+                and expand_sub_workflows
+        ):
+            sub_workflow_name = task
+            task_shards = workflow_metadata["calls"][sub_workflow_name]
+            print(f"{indent}SubWorkflow {sub_workflow_name}")
+
+            # For each element in total number of subworkflow calls get the subworkflow
+            # metadata. This loop will go through each shard if task is scattered
+            for i in range(len(task_shards) - 1):
+                sub_workflow_metadata = task_shards[i]["subWorkflowMetadata"]
+
+                print_workflow_status(
+                    workflow_metadata=sub_workflow_metadata,
+                    indent=indent + "\t",
+                    expand_sub_workflows=expand_sub_workflows,
+                )
+
+        # If no subworkflow is found then print status summary for task
+        else:
+            print_task_status(task, indent, workflow_metadata)
+
+
+def print_task_status(task: str, indent: str, workflow_metadata: dict) -> None:
+    """
+    Prints the task name and status count
+
+    :param task: Name of the task
+    :param indent: Indent string given as "\t", used to indent print out
+    :param workflow_metadata: Metadata of the workflow to process
+    :return:
+    """
+
+    shards = workflow_metadata["calls"][task]
+
+    shard_status_count = get_shard_status_count(shards)
+
+    shards_done = shard_status_count["Done"] if "Done" in shard_status_count else 0
+    shards_running = shard_status_count[
+        "Running"] if "Running" in shard_status_count else 0
+    shards_failed = shard_status_count[
+        "Failed"] if "Failed" in shard_status_count else 0
+    shards_retried = shard_status_count[
+        "RetryableFailure"] if "RetryableFailure" in shard_status_count else 0
+
+    # Determine what color to print task print out
+    if shards_failed == 0 and shards_running == 0:
+        task_status_font = io_utils.TextStatusesColor.TASK_COLOR_SUCCEEDED
+    elif shards_failed > 0 and shards_running > 0:
+        task_status_font = io_utils.TextStatusesColor.TASK_COLOR_FAILING  # Running but will fail
+    elif shards_running > 0:
+        task_status_font = io_utils.TextStatusesColor.TASK_COLOR_RUNNING
+    else:
+        task_status_font = io_utils.TextStatusesColor.TASK_COLOR_FAILED
+
+    print(
+        colored(
+            f"{indent}{task}\t{shards_running} Running, {shards_done} Done, {shards_retried} Preempted, {shards_failed} Failed",
+            color=task_status_font,
+        )
+    )
+
+    # If the task has shards that failed list them
+    # Maybe place contents of if statement in function and add indentation to printout??
```

Comment:
I agree with this comment

Reply:
ha, this must be my old comment. The contents of the if statements is already in a function. I'll remove the comment